editor/src/components/canvas/canvas-utils.ts

```diff
@@ -1149,78 +1139,211 @@ export function collectGuidelines(
+                guidelines.push({
+                  guideline: cornerGuideline(
                     point.x,
                     point.y + imageDimension.height,
                     imageDimension.width,
                     -imageDimension.height,
                   ),
-                )
+                  pointsOfRelevance: [
+                    canvasPoint({
+                      x: point.x,
+                      y: point.y + imageDimension.height,
+                    }),
+                  ],
+                })
               }
             } else {
               if (resizingFromPosition.y === 0) {
                 // Top-right.
-                guidelines.push(
-                  cornerGuideline(
+                guidelines.push({
+                  guideline: cornerGuideline(
                     point.x + imageDimension.width,
                     point.y,
                     -imageDimension.width,
                     imageDimension.height,
                   ),
-                )
+                  pointsOfRelevance: [
+                    canvasPoint({
+                      x: point.x + imageDimension.width,
+                      y: point.y,
+                    }),
+                  ],
+                })
               } else {
                 // Bottom-right.
-                guidelines.push(
-                  cornerGuideline(point.x, point.y, imageDimension.width, imageDimension.height),
-                )
+                guidelines.push({
+                  guideline: cornerGuideline(
+                    point.x,
+                    point.y,
+                    imageDimension.width,
+                    imageDimension.height,
+                  ),
+                  pointsOfRelevance: [
+                    canvasPoint({
+                      x: point.x,
+                      y: point.y,
+                    }),
+                  ],
+                })
               }
             }
           } else if (isEdgePositionAVerticalEdge(resizingFromPosition)) {
             // If this is a side edge the guidelines will be at x +/- width and y +/- (height / 2).
             guidelines.push(
-              xAxisGuideline(
-                point.x - imageDimension.width,
-                point.y - lowHalfHeight,
-                point.y + highHalfHeight,
-              ),
-              xAxisGuideline(
-                point.x + imageDimension.width,
-                point.y - lowHalfHeight,
-                point.y + highHalfHeight,
-              ),
-              yAxisGuideline(point.y - lowHalfHeight, point.x - imageDimension.width, point.x),
-              yAxisGuideline(point.y - lowHalfHeight, point.x, point.x + imageDimension.width),
-              yAxisGuideline(point.y + highHalfHeight, point.x - imageDimension.width, point.x),
-              yAxisGuideline(point.y + highHalfHeight, point.x, point.x + imageDimension.width),
+              {
+                guideline: xAxisGuideline(
```

Comment:
I just realized that there's a way to write these with no duplication (basically a function that returns the guideline + the points based on the guildelin endpoints), I can add it on in this PR or in one immediately after this one (so the diff is more understandable)

Reply:
I'd say add it in a separate PR to keep this one as clear as possible

Reply:
a follow-up PR sounds good!